packages/client/src/v2-events/features/events/AdvancedSearch/utils.ts

```diff
@@ -47,3 +55,135 @@ export const getAdvancedSearchFieldErrors = (
 
 export const flattenFieldErrors = (fieldErrors: Errors) =>
   Object.values(fieldErrors).flatMap((errObj) => errObj.errors)
+
+const defaultSearchFieldGenerator: Record<
+  // this should be strictly typed, not just any string
+  string,
+  (config: SearchField) => FieldConfig
+> = {
+  [EventFieldId.enum.trackingId]: (_) => ({
+    id: 'event.trackingId',
+    type: 'TEXT',
+    label: {
+      defaultMessage: 'Tracking ID',
+      description: 'Label for tracking ID field',
+      id: 'v2.advancedSearch.trackingId'
+    }
+  }),
+  [EventFieldId.enum.status]: (config) => ({
+    id: 'event.status',
+    type: 'SELECT',
+    label: {
+      defaultMessage: 'Status of record',
+      description: 'Label for status field',
+      id: 'v2.advancedSearch.status'
+    },
+    options: config.options ?? []
+  })
+}
+
+export const getDefaultSearchFields = (
+  section: AdvancedSearchConfig
+): FieldConfig[] => {
+  const searchFields: FieldConfig[] = []
+
+  section.fields.forEach((fieldConfig) => {
+    const generator = defaultSearchFieldGenerator[fieldConfig.fieldId]
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    if (generator) {
+      searchFields.push(generator(fieldConfig))
+    }
+  })
+
+  return searchFields
+}
+
+const RegStatus = {
+  Archived: 'ARCHIVED',
+  Certified: 'CERTIFIED',
+  CorrectionRequested: 'CORRECTION_REQUESTED',
+  DeclarationUpdated: 'DECLARATION_UPDATED',
+  Declared: 'DECLARED',
+  InProgress: 'IN_PROGRESS',
+  Issued: 'ISSUED',
+  Registered: 'REGISTERED',
+  Rejected: 'REJECTED',
+  Validated: 'VALIDATED',
+  WaitingValidation: 'WAITING_VALIDATION',
+  Created: 'CREATED'
+} as const
+
+type Condition =
+  | { type: 'fuzzy'; term: string }
+  | { type: 'exact'; term: string }
+  | { type: 'range'; gte: string; lte: string }
+  | { type: 'anyOf'; terms: string[] }
+
+export const ADVANCED_SEARCH_KEY = 'and'
+
+function buildCondition(type: string, value: string): Condition {
```

Comment:
If we do `export const MatchType = z.enum(['FUZZY', 'EXACT', 'RANGE'])` in `AdvancedSearchConfig.ts`, we can then type the function:

`function buildCondition(type: MatchType, value: string): Condition {`

Btw should the MatchType enum contain `ANY_OF`? I'm not sure.